src/frontends/tensorflow/src/input_model.cpp

```diff
@@ -346,6 +347,7 @@ std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::topologicall
                 if (m_tensor_places.find(producer_name) != m_tensor_places.end()) {
                     const auto& tensor_place = m_tensor_places[producer_name];
                     is_input |= tensor_place->is_input();
+                    m_found_inputs.insert(producer_name);
                 }
 
```

Comment:
please add test to cover this corner case

Reply:
Do you know how? I have no ideas

Reply:
Moreover, I think all three ways should be covered, but I don't know how to emulate this situations right now...

My suggestion: if this change should be merged before code freeze - create a task for tests and work on it separately. If I find a way how to test it earlier - add the test right here. @rkazants what do you think?

Reply:
Don't think it is complicated to simulate. Let us discuss this offline - I will tell you.